llama_cloud_services/parse/base.py

```diff
@@ -1120,41 +1122,35 @@ async def aparse(
                     file_names.append(str(f))
 
             try:
-                job_ids = await run_jobs(
-                    jobs,
-                    workers=self.num_workers,
-                    desc="Creating parsing jobs",
-                    show_progress=self.show_progress,
-                )
```

Comment:
Original issue is that this spawns all of the jobs at once (it batches the upload/parse API call, instead of starting up to `num_workers` parse jobs and then starting subsequent jobs when a previous one finishes)